app/models/canonical_armor.rb

```diff
@@ -1,11 +1,25 @@
 # frozen_string_literal: true
 
 class CanonicalArmor < ApplicationRecord
-  has_many :enchantments, through: :canonical_armors_enchantments
-  has_many :smithing_materials, through: :canonical_armors_smithing_materials, source: :canonical_materials
-  has_many :tempering_materials, through: :canonical_armors_tempering_materials, source: :canonical_materials
+  has_many :canonical_armors_enchantments, dependent: :destroy
+  has_many :enchantments,
+           -> { select 'enchantments.*, canonical_armors_enchantments.strength as enchantment_strength' },
```

Comment:
This enables us to call `item.enchantments.first.enchantment_strength` to get the `strength` value off the join table.